DOCS-2293, DOCS-3308, and DOCS-3156 Using Synthetic Test Monitors Guide (#13988)
DataDog/documentation: content/en/synthetics/guide/synthetic-test-monitors.md

---
title: Using Synthetic Test Monitors
kind: guide
description: Learn about Synthetic monitors created with your Synthetic tests. 
further_reading:
- link: "/monitors/manage/"
  tag: "Documentation"
  text: "Manage Monitors"
- link: "/synthetics/metrics/"
  tag: "Documentation"
  text: "Synthetic Monitoring Metrics"
---

## Overview

> When you create a Synthetic test, Datadog automatically creates an associated monitor. You are notified when the Synthetic test monitor alerts.

Reviewer: ```suggestion
When you create a Synthetic test, Datadog automatically creates an associated monitor. You are can setup notifications for when the Synthetic test monitor alerts.
```

> When you create a Synthetic test, Datadog automatically creates an associated monitor. You can setup notifications when the Synthetic test monitor alerts.

Reviewer 2: ```suggestion
When you create a Synthetic test, Datadog automatically creates an associated monitor. You can set up notifications when the Synthetic test monitor alerts.
```

{{< img src="synthetics/guide/synthetics_test_monitors/synthetic_test_monitor.png" alt="Synthetic test monitor" style="width:100%;">}}

## Create a Synthetic test monitor

<div class="alert alert-info">You cannot create or import a Synthetic test monitor in <a href="/monitors/create/">Monitors</a>.</div>

> Create a monitor in the **Configure the monitor for this test** section to send notifications when a Synthetic test is failing. Monitors are associated with the Synthetic test you create and contain the alerting conditions set in your Synthetic test configuration. To use monitor attribute and tag variables, create a [metric monitor][1].

Reviewer: ```suggestion
Create a monitor in the **Configure the monitor for this test** section to send notifications when a Synthetic test is failing. Monitors are associated with the Synthetic test you create and link to the alerting conditions set in your Synthetic test configuration. To use monitor attribute and tag variables, create a [metric monitor][1].
```

{{< img src="synthetics/guide/synthetics_test_monitors/configure_the_monitor_for_this_test.png" alt="Creating a monitor in your Synthetic test" style="width:90%;">}}

Customize the monitor name to search for it on the [**Manage Monitors**][2] page. To find a Synthetic test monitor, filter on `type:synthetics` in the search bar. You can use monitor [conditional variables][3] to characterize the notification message based on test state. 

The Synthetic test monitor integrates with notification channels such as email, Slack, Pagerduty, and Microsoft Teams. For more information, see [Notifications][4].

If you have multiple layers of notifications (for example, notifying more teams the longer a Synthetic test is alerting), Datadog recommends enabling [renotification][5] on your Synthetic monitors.

## Tailor monitor notifications

> Depending on your incident management strategy, you may want to involve multiple teams when a Synthetic test alerts. Add `{{#is_renotify}}` and `{{/is_renotify}` in the escalation message to notify Team A first, and Team B if the issue still occurs. Use [conditional variables][3] to characterize the notification message based on team. 

Reviewer 2: ```suggestion
Depending on your incident management strategy, you may want to involve multiple teams when a Synthetic test alerts. To notify Team B only on subsequent alerts after the first alert, surround the notification to Team B with `{{#is_renotify}}` and `{{/is_renotify}`. Use [conditional variables][3] to further characterize the notification message based on monitor attributes. 
```
This line is fine as is, but could be reworded to be clearer about how exactly you write the message to notify a different team upon escalation.

{{< img src="synthetics/guide/synthetics_test_monitors/renotification_toggle.png" alt="Select the amount of time for the alerting monitor to renotify" style="width:90%;">}}

> To enable the alerting monitor to renotify, click the toggle left of `If this monitor stays in alert status renotify every` and select a time option from the dropdown menu. Optionally, use [conditional variables][3] such as `{{#is_renotify}}` and `{{/is_renotify}}` to send an escalation message to a different team or organization.

Reviewer 2: ```suggestion
To enable the alerting monitor to renotify, click the toggle left of `If this monitor stays in alert status renotify every` and select a time option from the dropdown menu.
```
I deleted this line because it duplicates information from above the image.

---
title: Using Synthetic Test Monitors
kind: guide
description: Learn about Synthetic monitors created with your Synthetic tests. 
further_reading:
- link: "/monitors/manage/"
  tag: "Documentation"
  text: "Manage Monitors"
- link: "/synthetics/metrics/"
  tag: "Documentation"
  text: "Synthetic Monitoring Metrics"
---

## Overview

When you create a Synthetic test, Datadog automatically creates an associated monitor. You are notified when the Synthetic test monitor alerts.

> {{< img src="synthetics/guide/synthetics_test_monitors/synthetic_test_monitor.png" alt="Synthetic test monitor" style="width:100%;">}}

Reviewer: Can we change this image to this one (so Synthetics -> Monitors versus Monitors -> Synthetics): https://share.getcloudapp.com/yAuKOoKW (let me know if we need a new screenshot)

## Create a Synthetic test monitor

<div class="alert alert-info">You cannot create or import a Synthetic test monitor in <a href="/monitors/create/">Monitors</a>.</div>

> Create a monitor in the **Configure the monitor for this test** section to send notifications when a Synthetic test is failing. Monitors are associated with the Synthetic test you create and set alerting conditions for. 

Reviewer: ```suggestion
Create a monitor in the **Configure the monitor for this test** section to send notifications when a Synthetic test is failing. Monitors are associated with the Synthetic test you create and driven by the alerting conditions you set within your Synthetic test configuration. 
```
Let me know what you think of the wording.

{{< img src="synthetics/guide/synthetics_test_monitors/configure_the_monitor_for_this_test.png" alt="Creating a monitor in your Synthetic test" style="width:90%;">}}

> Customize the monitor name to search for it on the [**Manage Monitors**][1] page. You can use [conditional variables][2] to characterize the notification message based on test state. To find a Synthetic test monitor, filter on `type:synthetics` in the search bar.

Reviewer: ```suggestion
Customize the monitor name to search for it on the [**Manage Monitors**][1] page. To find a Synthetic test monitor, filter on `type:synthetics` in the search bar. You can use Monitor [conditional variables][2] to characterize the notification message based on test state. 
```

The Synthetic test monitor integrates with notification channels such as email, Slack, Pagerduty, and Microsoft Teams. For more information, see [Notifications][3].

> Synthetic test monitors do not support tag and attribute variables or the `{{#is_warning}}`, `{{^is_warning}}`, `{{#is_warning_recovery}}`, and `{{^is_warning_recovery}}` conditional variables in the the monitor name. To use tag and attribute variables, create a [metric monitor][4].

Reviewer: Did we want to remove the explicit variables we don't support? And just have something like:

```suggestion
To use Monitor attribute and tag variables, create a [metric monitor][4].
```

And do we want to link to the separate guide for using the `synthetics.test_runs` metric?

Author: Yes, we will do the latter in DOCS-3513.

> If you have multiple layers of notifications (for example, with warning thresholds), Datadog recommends enabling [renotification][5] on your monitors.

Reviewer: ```suggestion
If you have multiple layers of notifications (for example, notifying multiple teams the longer a Synthetic test is in an alerting state), Datadog recommends enabling [renotification][5] on your Synthetic monitors.
```

---

> title: Understanding Synthetic Tests and Alerting Monitors

Reviewer 3: - Any reason why we are talking about `Alerting Monitors` and not just `Monitors` here? 
- Also, it feels like this doc is not really only about "Understanding" the relationship between the two objects, right? Several sections feel more like tips on how to use monitors to answer some specific use cases, wdyt?

Reviewer: "Tips and tricks to benefit from your Synthetic test monitor"? Lol might be too much

kind: guide
description: Learn about Synthetic monitors created with your Synthetic tests. 
further_reading:
- link: "/monitors/manage/"
  tag: "Documentation"
  text: "Manage Monitors"
- link: "/synthetics/metrics/"
  tag: "Documentation"
  text: "Synthetic Monitoring Metrics"
---

## Overview

> When you create a Synthetic test, Datadog automatically creates an associated monitor. The monitor's alerting conditions in your test configuration impact how you are notified when a Synthetic test moves into an alerting state.

Author: Sounds good, I'm adding this entry in the next commit.

{{< img src="synthetics/guide/synthetics_test_monitors/synthetic_test_monitor.png" alt="Synthetic test monitor" style="width:100%;">}}

> This guide describes how Synthetic test monitors behave, how to use a monitor's conditional variables in a Synthetic test configuration, and how to set downtimes and mute monitors.

Reviewer 3: Might need to revamp this sentence slightly? It does not seem to comprehensively reflect the content of the guide + it mentions downtimes that I don't see mentioned below?

> ## Create a Synthetic test monitor

Reviewer 3: Why doesn't this section live in https://docs.datadoghq.com/synthetics/api_tests/http_tests?tab=requestoptions#notify-your-team and in all individual test pages? Or in both test pages and this guide? Also, there does not seem to be a link between the test individual pages and this guide, is that normal?

Author: Good question! I was planning on publishing this guide first and doing a sweep through the individual API test pages second. We can collaborate on linking the two pieces of content together 😄

Author: Created DOCS-3508.

> <div class="alert alert-info">Alerting monitors created by Synthetic tests are not a <a href="/monitors/create/">monitor type</a>. You cannot create or import a Synthetic test monitor.</div>

Reviewer: _Technically_ they are a monitor type, they are just very unique/we don't surface them within the monitor creation workflow (which is why you cannot create/import them).

Reviewer 3: +1, users can filter on `type:synthetics` on the `Manage Monitors` page.

Following the test creation workflow, create a monitor in the **Configure the monitor for this test** section to send notifications when a Synthetic test is failing. The Synthetic test monitor integrates with notification channels such as email, Slack, Pagerduty, and Microsoft Teams.

{{< img src="synthetics/guide/synthetics_test_monitors/configure_the_monitor_for_this_test.png" alt="Creating a monitor in your Synthetic test" style="width:90%;">}}

Customize the monitor name to identify it on the [**Manage Monitors**][1] page. You can also use [conditional variables][2] to characterize the notification message based on test state.

> Synthetic test monitors do not support `{{#is_warning}}`, `{{^is_warning}}`, `{{#is_warning_recovery}}`, and `{{^is_warning_recovery}}` conditional variables. Datadog recommends enabling [renotification][3] on your monitors.

Reviewer 3: 1. I'm not sure I understand why there is a mention of what is not supported here? These conditional variables do not appear in the autocompletion dropdown so I would be in favor of not mentioning these warning conditional variables at all, WDYT?
2. Why saying we recommend using renotifications? Whether or not to use it really depends on users' needs IMO but I might be missing something here!

> Synthetic test monitors do not support `{{#is_warning}}`, `{{^is_warning}}`, `{{#is_warning_recovery}}`, and `{{^is_warning_recovery}}` conditional variables. Datadog recommends enabling [renotification][3] on your monitors.

Reviewer: Some customers look at the monitor documentation and expect to see the same conditional variables for Synthetic monitor config, so it was to highlight that. The recommendation to use renotifications were in the case of needing two layers of notifications (traditionally through warning thresholds).

It does increase the complexity of the guide I admit 🤔

> Synthetic test monitors do not support `{{#is_warning}}`, `{{^is_warning}}`, `{{#is_warning_recovery}}`, and `{{^is_warning_recovery}}` conditional variables. Datadog recommends enabling [renotification][3] on your monitors.

Author: Tag and attribute variables are also not available through Synthetic monitors. To use tag and attribute variables, use a Metric Monitor instead. + Link to Use Synthetic test metrics header.

> ## Notify multiple teams with monitor renotifications

Reviewer 3: I'm not sure I understand this section, what are we saying here? That users can use the `{{#is_renotify}}{{/is_renotify}} ` escalation message in order to notify team A first, then team B if issue is still occurring? I feel like there's a discrepancy between the title and the content of the section 😬.

> Depending on your incident management strategy, you may want to involve multiple teams when a Synthetic test alerts. You can use the available conditional variables to set which team are notified in your notification message. 

Reviewer 3: We might want to clarify the link between conditional variables and notifying multiple teams here, WDYT?

Author: Added link to /monitors/notify/variables/?tab=is_alert#conditional-variables.

> To enable renotifications, click the toggle left of `If this monitor stays in alert status renotify every` and select a time option from the dropdown menu.

Reviewer: Should we include this screenshot here: https://a.cl.ly/Bluo9EP6

Reviewer 3: It might be worth clarifying the link between this setting and the `{{#is_renotify}}{{/is_renotify}}` variables.

> ## Integrate your Synthetic test monitor with Statuspage

Reviewer 3: YES! So happy you worked on that!
Why not creating a standalone page out of it (maybe not now, but later on)? I feel like it's a use case of its own and I'm afraid it might be a bit hidden in here. I think we discussed it @peculiarism but if we had a more general page, we could make it monitor type agnostic and link it from https://docs.datadoghq.com/monitors/guide/ as well!

Reviewer: Great point about making it monitor agnostic!

Author: Created DOCS-3509 to work on this later!

If you use [Atlassian Statuspage][4] for visibility into your applications' and services' uptime, you can update the status of your systems with  Synthetic test monitor notifications.

1. See the [Statuspage documentation][5] to generate a component-specific email address.
2. Add the generated email address into your test's notification message.

> 3. Customize the monitor name to return `UP` or `DOWN` depending on the test state.

Reviewer 3: You might want to mention conditional variables here to make it clear to users how they can get this working without having them look into the screenshot.

{{< img src="synthetics/guide/synthetics_test_monitors/statuspage_monitor_setup.png" alt="Add a Statuspage email address and status to the monitor name in your Synthetic test" style="width:95%;">}}

> ## Use Synthetic test metrics

Author: Agreed, this could be its own guide topic, and we can dive more in depth of examples.

> ## Use Synthetic test metrics

Author: Scoped in DOCS-3513.

> You may want to group monitor notifications based on test run status or team. Use [Synthetic Monitoring Metrics][6] to fine tune your notification capabilities and individual test notifications by creating additional [metric monitors][7].

Reviewer 3: I think the first sentence needs to be clarified, it is unclear what the capability you are mentioning here would allow users to do.
Also by saying fine tuning, it sounds a bit like users have to edit their Synthetic monitors (which contradicts the rest of the sentence where you explain a standalone metric monitor needs to be created). Also not sure I understand what you mean by "individual test notifications"?

Why not adding an example to make it clearer what the recommended set up is here?

Author: Sounds good, I'm going to remove this section + created DOCS-3513.

## Export Synthetic test uptime

> In order to export the uptime for a Synthetic test, you need the `monitor_id` for the respective test. Use the [Monitors API][8] to query the [`Get an API test`][9] or [`Get a browser test` endpoint][10].

Reviewer: @MargotLepizzera yes, we get requests for the ability to extract the uptime widget values for individual tests. I'll include a thorough example here of which parameters are needed
